e2e_tests/tests/cluster/test_users.py

```diff
@@ -266,61 +265,107 @@ def test_logout(clean_auth: None) -> None:
+
+    # Now log in again.
+    assert log_in_user(creds) == 0
+
+    # SDK testing for activating and deactivating.
+    log_in_user(ADMIN_CREDENTIALS)
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=creds.username)
+    assert not user.deactivate().user.active
+    assert user.activate().user.active
 
     # Now log in again.
     assert log_in_user(creds) == 0
 
 
 @pytest.mark.e2e_cpu
 @pytest.mark.e2e_cpu_postgres
-def test_change_password(clean_auth: None) -> None:
+def test_change_password(clean_auth: None, login_admin: None) -> None:
     # Create a user without a password.
-    creds = create_test_user(ADMIN_CREDENTIALS, False)
+    creds = create_test_user(False)
 
     # Attempt to log in.
     assert log_in_user(creds) == 0
 
     # Log out.
     assert log_out_user() == 0
 
-    newPassword = get_random_string()
-    assert change_user_password(creds.username, newPassword, ADMIN_CREDENTIALS) == 0
+    # login admin
+    log_in_user(ADMIN_CREDENTIALS)
 
+    newPassword = get_random_string()
+    assert change_user_password(creds.username, newPassword) == 0
     assert log_in_user(authentication.Credentials(creds.username, newPassword)) == 0
 
+    newPasswordSdk = get_random_string()
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=creds.username)
+    assert user.change_password(new_password=newPasswordSdk)
+    assert log_in_user(authentication.Credentials(creds.username, newPasswordSdk)) == 0
+
+
+@pytest.mark.e2e_cpu
+def test_change_username(clean_auth: None, login_admin: None):
+    creds = create_test_user()
+    new_username = "renameuser4"
+    print("new username")
+    print(creds.username)
+    print(new_username)
+    command = ["det", "-m", conf.make_master_url(), "user", "rename", creds.username, new_username]
+    assert subprocess.call(command) == 0
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=new_username)
+    assert user.username == new_username
+    assert log_in_user(authentication.Credentials(new_username, "")) == 0
+
+    # Test SDK
+    new_username = "renameuser5"
+    user.rename(new_username)
+    user = det_obj.get_user_by_name(user_name=new_username)
+    assert user.username == new_username
+    assert log_in_user(authentication.Credentials(new_username, "")) == 0
+
 
 @pytest.mark.e2e_cpu
 @pytest.mark.e2e_cpu_postgres
 @pytest.mark.e2e_cpu_cross_version
-def test_experiment_creation_and_listing(clean_auth: None) -> None:
+def test_experiment_creation_and_listing(clean_auth: None, login_admin: None) -> None:
     # Create 2 users.
-    creds1 = create_test_user(ADMIN_CREDENTIALS, True)
+    creds1 = create_test_user(True)
 
-    creds2 = create_test_user(ADMIN_CREDENTIALS, True)
+    creds2 = create_test_user(True)
+
+    # Ensure determined creds are the default values.
+    change_user_password("determined", "")
```

Comment:
why do this here?

Reply:
experiment.run_basic_test is using the determined credentials to run an exp so this is to ensure it's the right credentials in case the password of determined was changed somewhere else.